regexp.go

```diff
@@ -65,67 +65,88 @@ func newRouteRegexp(tpl string, typ regexpType, options routeRegexpOptions) (*ro
 	}
 	varsN := make([]string, len(idxs)/2)
 	varsR := make([]*regexp.Regexp, len(idxs)/2)
-	pattern := bytes.NewBufferString("")
+
+	var pattern, reverse strings.Builder
 	pattern.WriteByte('^')
-	reverse := bytes.NewBufferString("")
-	var end int
+
+	var end, colonIdx, groupIdx int
 	var err error
+	var patt, param, name string
 	for i := 0; i < len(idxs); i += 2 {
 		// Set all values we are interested in.
+		groupIdx = i/2
+
 		raw := tpl[end:idxs[i]]
 		end = idxs[i+1]
-		parts := strings.SplitN(tpl[idxs[i]+1:end-1], ":", 2)
-		name := parts[0]
-		patt := defaultPattern
-		if len(parts) == 2 {
-			patt = parts[1]
+
+		param = tpl[idxs[i]+1 : end-1]
+		colonIdx = strings.Index(param, ":")
+		if colonIdx == -1 {
+			name = param
+			patt = defaultPattern
+		} else {
+			name = param[0:colonIdx]
+			patt = param[colonIdx+1:]
+		}
+		if patt == "" {
+			patt = defaultPattern
 		}
+
 		// Name or pattern can't be empty.
 		if name == "" || patt == "" {
-			return nil, fmt.Errorf("mux: missing name or pattern in %q",
-				tpl[idxs[i]:end])
+			return nil, fmt.Errorf("mux: missing name or pattern in %q", param)
```

Comment:
```suggestion
			return nil, fmt.Errorf("mux: missing name or pattern in %q", tpl[idxs[i]:end])
```
Having `param` as the variable might not help much, as when there's no pattern or name inside braces {}, param will just return an empty string (""), whereas. Above will at least help with the braces for reference.

Reply:
Gotcha, so {} would be printed as well. Sounds legit, I'll give it a once over and add a test, some inline comments, it's a bit hard to read after some time since writing it